components/managedidentity/04-managed-identities.tf

```diff
@@ -51,6 +52,60 @@ resource "azurerm_cosmosdb_sql_role_assignment" "identity_contributor" {
   scope              = data.azurerm_cosmosdb_account.version_reporter.id
 }
 
+data "azurerm_cosmosdb_account" "pipeline_metrics" {
+  count               = var.env == "prod" || var.env == "sandbox" ? 1 : 0
+  provider            = azurerm.managed_identity_infra_subs
+  name                = local.mi_cft[local.jenkins_mi_environment].cosmosdb_name
+  resource_group_name = local.mi_cft[local.jenkins_mi_environment].resource_group_name
+}
+
+resource "azurerm_cosmosdb_sql_role_assignment" "monitoring_mi_assignment" {
+  count               = var.env == "prod" || var.env == "sandbox" ? 1 : 0
+  provider            = azurerm.ptl
+  resource_group_name = data.azurerm_cosmosdb_account.pipeline_metrics[count.index].resource_group_name
+  account_name        = data.azurerm_cosmosdb_account.pipeline_metrics[count.index].name
+  role_definition_id  = "${data.azurerm_cosmosdb_account.pipeline_metrics[count.index].id}/sqlRoleDefinitions/00000000-0000-0000-0000-000000000002"
+  principal_id        = azurerm_user_assigned_identity.managed_identity.principal_id
+  scope               = data.azurerm_cosmosdb_account.pipeline_metrics[count.index].id
+}
+
+data "azuread_service_principals" "pipeline" {
+  display_names = [
+    "DTS Bootstrap (sub:dcd-cftapps-sbox)",
+    "DTS Bootstrap (sub:dcd-cftapps-dev)",
+    "DTS Bootstrap (sub:dcd-cftapps-ithc)",
+    "DTS Bootstrap (sub:dcd-cftapps-demo)",
+    "DTS Bootstrap (sub:dcd-cftapps-stg)",
+    "DTS Bootstrap (sub:dcd-cftapps-test)",
+    "DTS Bootstrap (sub:dcd-cftapps-prod)",
+    "DTS Bootstrap (sub:dts-cftsbox-intsvc)",
+    "DTS Bootstrap (sub:dts-cftptl-intsvc)"
+  ]
+}
+
+resource "azurerm_role_assignment" "rbac_admin" {
+  provider             = azurerm.ptl
+  for_each             = length(data.azurerm_cosmosdb_account.pipeline_metrics) > 0 ? { for sp in data.azuread_service_principals.pipeline.service_principals : sp.object_id => sp } : {}
+  role_definition_name = "DocumentDB Account Contributor"
+  principal_id         = each.value.object_id
+  scope                = length(data.azurerm_cosmosdb_account.pipeline_metrics) > 0 ? data.azurerm_cosmosdb_account.pipeline_metrics[0].id : null
+}
```

Comment:
I also mentioned that this part of things makes more sense to live in the jenkins-infra repo side -- where the service connections built and own those cosmos accounts

The changes to this repo should only be adding a role assignment to all our monitoring MIs